site/docs/plugins/entity-parser.md

```diff
@@ -0,0 +1,249 @@
+---
+prev: false
+next: false
+---
+
+# Entity Parser (`entity-parser`)
+
+Converts [Telegram entities](https://core.telegram.org/bots/api#messageentity)
+to semantic HTML.
+
+## When Should I Use This?
+
+Probably NEVER!
+
+While this plugin can generate HTML, it's generally best to send the text and entities back to Telegram.
+
+Converting them to HTML is only necessary in rare cases where you need to use Telegram-formatted text **outside** of Telegram itself, such as displaying Telegram messages on a website.
+
+See the [_Cases When It's Better to Not Use This Package_](#cases-when-it-s-better-to-not-use-this-package) section to determine if you have a similar problem to solve.
+
+If you're unsure whether this plugin is the right fit for your use case, please don't hesitate to ask in our [Telegram group](https://t.me/grammyjs).
+In most cases, people find they don't actually need this plugin to solve their problems!
```

Comment:
It is already made clear above that the plugin is not needed most of the times.

```suggestion
```

Reply:
Can we keep the offer to help?
```suggestion
If you're unsure whether this plugin is the right fit for your use case, please don't hesitate to ask in our [Telegram group](https://t.me/grammyjs).
```

Reply:
> Can we keep the offer to help?

IMO yes.

Also, it's good to have a big fat warning. However, it now is not quite clear what the plugin actually does.

In other words, when reading this, I have now understood that I _probably_ don't want to use it. When _should_ I use it? This needs to be improved.

In my opinion, the main source of confusion is that the plugin never actually parses anything. It generates HTML, so it should rather be called entity-renderer or maybe even include HTML in the name somewhere.

Reply:
> so it should rather be called entity-renderer or maybe even include HTML in the name somewhere

`entity-to-html`, perhaps?

> In other words, when reading this, I have now understood that I probably don't want to use it. When should I use it? This needs to be improved.

This plugin simply renders entities into HTML, yes. However, my concern is that people might misuse it to convert entities, make edits, and then send them back to Telegram, whereas they should be working with the entities directly. Maybe I need some help to rephrase this for the doc more effectively.

Reply:
I understand the concern and I actually share it, too. You say this:

> is only necessary in rare cases where you need to use Telegram-formatted text **outside** of Telegram itself

And that's a good explanation, but it needs more structure. The current docs effectively say

- don't use it
- use it for entities outside of TG
- don't use it without asking

and I feel like it could be made a little clearer. But maybe I'm just tired. I'll try to suggest an improvement tomorrow if you feel like my words still don't really make sense.